be/src/olap/data_dir.cpp

```diff
@@ -373,6 +375,14 @@ Status DataDir::load() {
                                     const std::string& meta_str) -> bool {
         RowsetMetaSharedPtr rowset_meta(new RowsetMeta());
         bool parsed = rowset_meta->init(meta_str);
+        // convert delete predicate to v2 (#22442)
+        if (rowset_meta->has_delete_predicate()) {
```

Comment:
need write back and then we could delete these code in 2.2

Reply:
ok, actually I'm thinking how to write back in one traverse with least change, our rowset meta traverse is under read semantics.